openms-wiki/developer-faq.md

```diff
@@ -0,0 +1,491 @@
+
+On Windows, Visual Studio solution files are automatically build with the `/MP` flag, such that Visual Studio uses all available cores of the machine.
+
+## Release
+
+View [Preparation of a new OpenMS release](https://github.com/OpenMS/OpenMS/wiki/Preparation-of-a-new-OpenMS-release#release_developer) to learn more about contributing to releases.
+
+## Continuous integration
+
+### How does travis work?
+
+[Travis](https://travis-ci.org/) is an automated system for continuous integration. Each new commit and pull request is automatically run through the travis build system.
+This is controlled by a `.travis.yaml` file in the source tree.
+
+### What can we do if travis times out?
+
+Try to restart travis, it sometimes hangs and since it builds on shared infrastructure, the next build may work better. This needs to be done by a OpenMS core developer.
+
+Since we use extensive caching, the build may take much longer when many files are touched and may never complete in that case (running into the travis time limit). In that case we can rebuild the cache using the following approach:
+```bash
+ $ git cherry-pick 89c5cd7f2d9d343b3d63fc6bab18e08dcd969c05
+
+ $ git push origin develop
+
+ # Now wait for the build to complete
+
+ $ git revert 89c5cd7f2d9d343b3d63fc6bab18e08dcd969c05
+
+ $ git push origin develop
+```
+
+## Working in Integrated Development Environments (IDEs)
+
+### Why are there no `source/TEST` and `source/APPLICATIONS/TOPP|UTILS` folder?
+
+All source files added to an IDE are associated with their targets. You can find the source files for each test within its own subproject. The same is true for the `TOPP` and `UTILS` classes.
+
+### How can I easily update a lot of test files (e.g., after a small format change in result files)?
+
+Using `grep` one can simply extract the lines starting with `diff FILENAME1 FILENAME2` and replace the `diff` by `copy`.
+
+### I'm getting the error "Error C2471: cannot update program database"
+
+This is a bug in Visual Studio and there is a [bug fix] (http://code.msdn.microsoft.com/KB946040) Only apply it if you encounter the error. The bug fix might have unwanted side effects!
+
+### Visual Studio can't read the clang-format file.
+
+Depending on the Visual Studio version you're using you might get an error like `Error while formating with ClangFormat`. This is because Visual Studio is using an outdated version of clang-format. Unfortunately there is no easy way to update this using Visual Studio itself.
+There is a plugin provided by LLVM designed to fix this problem, but the plugin doesn't work with every Visual Studio version. However, you can update clang-format by hand using the pre-build clang-format binary. Both the binary and a link to the plugin can be found [here](https://llvm.org/builds/).
```

Comment:
> you can update clang-format by hand using the pre-build clang-format binary. Both the binary and a link to the plugin can be found

In that case, update `clang-format` manually using the pre-build clang-format binary.